src/components/App/App.jsx

```diff
@@ -91,18 +121,38 @@ function App() {
   };
 
   // Add selected card to favorite array
-  const handleAddFavorite = (recipe) => {
-    recipesList.find((item) => item._id === recipe._id).isFavorite = true;
-    setFavoriteList([recipe, ...favoriteList]);
+  const handleAddFavorite = (recipeId) => {
+    addFavoriteRecipe(recipeId)
+      .then((response) => {
+        setFavoriteList(getFavoriteList(response));
+      })
+      .catch((err) => {
+        console.error(err);
+      });
   };
 
   // Remove selected card from favorite array
-  const handleDeleteFavorite = (recipe) => {
-    recipesList.find((item) => item._id === recipe._id).isFavorite = false;
-    const tempFavorites = favoriteList.filter(
-      (item) => item._id !== recipe._id
-    );
-    setFavoriteList(tempFavorites);
+  const handleDeleteFavorite = (recipeId) => {
+    deleteFavoriteRecipe(recipeId)
+      .then((response) => {
+        setFavoriteList(getFavoriteList(response));
+      })
+      .catch((err) => {
+        console.error(err);
+      });
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors